### DISCUSSION TITLE
Understanding the usage of ivp_solve in the transient reactive transport module

### DISCUSSION THREAD

**Discussion by oscarhx** [PMEAL/OpenPNM#2522] [Q&A] [answered]
Hello, Profs. 

scipy's ivp_solve module was applied in the dev version of PNM. 
My question is that in the '_transient_reactive_transport.py' file, the def _build_rhs(self) function, the ode_func(t,y):
the return of (-A.dot(y)+b)/V. 
To my knowledge the return value equals the derivative of concentration by time, in situation of the fickian diffusion simulation,
which according to the transient diffusion equation the form of the return expression should represent the second derivative of concentration by spacing, in even lattice. I can't understand how the return expression in form of (-A.dot(y)+b)/V works for that purpose. 

Best,
Xiang

**Comment by ma-sadeghi** [comment, marked as answer]
Sorry for the late reply.

The A matrix represents the discretization. For example, for diffusion, that the $i$-th row of $Ax-b$ is the discretized mass balance for pore $i$. Now, since the balance equations we use in OpenPNM are not per volume (ex. the unit of the mass balance in OpenPNM is kg/s or mol/s, instead of kg/m3/s), the transient term is defined as
$$dm/dt = V \cdot dc/dt$$
So, to get the concentration over time, we need to divide the RHS by volume, hence `(A.dot(y)-b)/V`. 

I hope that's a bit more clear. Please let me know if that answers your question.

**Reply by oscarhx** [reply]
Hello Professor Sadehgi,
Your equation solved my problem.
Thanks
